common/src/main/java/com/wynntils/features/trademarket/TradeMarketQuickSearchFeature.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.trademarket;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.consumers.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.mc.event.MenuEvent.MenuClosedEvent;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.event.ScreenOpenedEvent;
+import com.wynntils.models.containers.containers.TradeMarketContainer;
+import com.wynntils.utils.mc.KeyboardUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraft.world.inventory.Slot;
+import net.neoforged.bus.api.EventPriority;
+import net.neoforged.bus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.TRADEMARKET)
+public class TradeMarketQuickSearchFeature extends Feature {
+    @Persisted
+    public final Config<Boolean> instantSearch = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> hidePrompt = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> autoCancel = new Config<>(true);
```

Comment:
Honestly I don't think any of these are necessary. Instant search should just be always on, I can't imagine a scenario where someone would want to quick search something but not have it be instant and require pressing enter to send the chat message.

If the message is sent instantly the prompt can always be hidden and you'll never need to cancel sending the search query

Reply:
Yeah it's the default, the shift to bring up the chat is what the other two options were originally for, but it was easy to add the option so I did for anyone who wants it, but ur right not a lot of people will use it

Reply:
Maybe use the hide prompt and auto cancel from the auto open chat feature? But that's also kinda janky

Reply:
Removing an option no one would use anyway (instant search off) I guess, tho why not leave it in as an option. But while this is not all in the model the other two are somewhat important. They were the reason I made the same option for "auto open chat"

Reply:
Okay, we added auto cancel and hide prompt in the other feature. Why would we want to duplicate the same thing here? 

I've now checked the code and I realized that this and `TradeMarketAutoOpenChatFeature` inheritedly wants to be the same feature. Please merge the two into `TradeMarketQuickSearchFeature` (admittedly, I like your feature name better here). That'll also get rid of the duplicate configs. Me or @ShadowCat117 can help with a config upfixer for persisting the enabled state of the old feature. Finally, sure, let's, keep `instantSearch`.

Reply:
Also, as Shadowcat mentioned in other places, the original feature is more refined, so please try to take everything over, and use that feature as a base, and extend this feature's changes into it.

Reply:
Okay if you put it that way, yeah I guess the whole opening chat automatically feature could be moved to Quick Search.
I just removed the options here as @ShadowCat117 was saying because I realized he was 100% correct because the options just worked out of the box even if they were disabled in QS.

Reply:
It's all in Quick Search now :)